skepticoin/scripts/utils.py

```diff
@@ -107,6 +107,7 @@ def start_networking_peer_in_background(
     port: Optional[int] = None if args.dont_listen else args.listening_port
     thread = NetworkingThread(coinstate, port)
     thread.local_peer.network_manager.disconnected_peers = load_peers()
+    thread.daemon = True
```

Comment:
I don't remember exactly why, but I chose a different idiom for closing the various threads. See e.g. `repl.py` for how it was done. Feel free to copy/paste that approach... though factoring it out is even better I suppose.

Now, if you can explain to me why the `thread.daemon` way is better, I might be persuaded to go that route, but in that case it should be in a separate PR (which includes said explanation) and should also be extended to all other locations where we currently have the try/finally approach from `repl.py`

Reply:
I'll move the `thread.daemon = True` to `balance.py`. That's where it's actually required.

Reply:
Without this, the balance.py just hangs.

Reply:
What about using `thread.stop()` and `thread.join()`, similar to ~what we do~ what we [will be doing](https://github.com/skepticoin/skepticoin/pull/35/files#diff-67cc9641a1f9cd51b7bf7e4d5758ef7d155a517cfb13be11032dfd4b1480b6bcR147-R151) in MinerWatcher?

Reply:
That could work too

Reply:
changed it in 6d3a66274798b77d6d38c5b73bc397efea25c01b